index.js

```diff
@@ -2,105 +2,82 @@
 'use strict';
 const Funnel = require('broccoli-funnel');
 const getDebugMacros = require('./src/debug-macros').debugMacros;
-const semver = require('semver');
-const pkg = require('./package.json');
-
-/*
- shouldIncludeChildAddon runs during
- addon init so does not have access to
- the host application instance.
-*/
-let requiredDataPackages = ['@ember-data/model', '@ember-data/store', 'ember-inflector'];
-let allDataPackages = [
-  '@ember-data/adapter',
-  '@ember-data/debug',
-  '@ember-data/model',
-  '@ember-data/record-data',
-  '@ember-data/serializer',
-  '@ember-data/store',
-];
-let minVersionForUsingOwnPackages = '3.14.0';
-
-function hasDataPackage(addon) {
-  let addons = addon.project.addonPackages;
-  return addons['ember-data'] !== undefined;
-}
-function hasOwnDataPackages(addon) {
-  let addons = addon.project.addonPackages;
 
-  for (let i = 0; i < allDataPackages.length; i++) {
-    if (addons[allDataPackages[i]] !== undefined) {
-      return true;
-    }
-  }
-  return false;
-}
-function assertOwnDataPackagesValid(addon) {
-  let addons = addon.project.addonPackages;
-
-  for (let i = 0; i < requiredDataPackages.length; i++) {
-    let requiredPackageName = requiredDataPackages[i];
-    if (addons[requiredPackageName] === undefined) {
-      throw new Error(
-        `ember-m3 ${pkg.version} requires the peerDependency ${requiredPackageName} to be installed.`
-      );
+const VersionChecker = require('ember-cli-version-checker');
+
+function assertValidEmberData(addon) {
+  let checker = VersionChecker.forProject(addon.project);
+  let emberDataFound = false;
+  let emberDataVersionsFound = [];
+  let emberDataModelFound = false;
+  let emberDataModelVersionsFound = [];
+  let emberDataStoreFound = false;
+  let emberDataStoreVersionsFound = [];
+  let emberInflectorFound = false;
+  let emberInflectorVersionsFound = [];
+
+  for (let { name, pkg } of checker.allAddons()) {
```

Comment:
we'll need to either fix the project wide check in ember-cli-version-checker or bring the replacement from the other PR over to this one.